docs/developers/CwlParsing.md

```diff
@@ -0,0 +1,40 @@
+# How CWL is Parsed
+
+The process of parsing CWL runs approximately like this:
+
+1. Produce a canonical one-file representation of the CWL as a JSON object in memory 
+1. Use Circe to turn that representation into scala case classes
+1. Convert the case classes into an appropriate WOM representations
+
+## Getting a canonical flat file
+
+### From a source file and dependencies zip
+
+The process looks like:
+
+1. Write the workflow source file to disk
+1. Make sure the dependencies are unzipped next to it
+1. `SALAD and flatten`:
+    - Run `cwltool --print-pre` against the source file to get a canonicalize JSON representation (aka `SALAD`) it.
```

Comment:
This is more of a detail and feel free to omit it here, just thought I'd mention it in case:
A CWL file can contain several workflows and tools. In that case a workflow root needs to be provided (either as a parameter of the request for `server` mode or an CLI option in `run` mode).
The whole source file will be saladed but the workflow root will be the starting point from which the flattening will begin. Meaning the end result will be a single flat workflow (the root workflow), even if the original file contained other workflows or tools.
You can see that in this unit test: 
[Original cwl](https://github.com/broadinstitute/cromwell/blob/develop/cwl/src/test/resources/cwl/preprocessor/complex_workflow/root_workflow.cwl)
[Flattened version with `echo-workflow-2` as root](https://github.com/broadinstitute/cromwell/blob/develop/cwl/src/test/resources/cwl/preprocessor/complex_workflow/expected_result.json)

Notice that the flattened version is a single workflow (in which references, including to other workflows in the same original file, have been inlined).